src/c++/perf_analyzer/docs/examples/profile.py

```diff
@@ -175,73 +223,134 @@ def collect_periodic_latencies(args, filename):
-        first_token_latencies.append(first_response - request["timestamp"])
-        prev_response = first_response
-        for response in remaining_responses:
-            token_to_token_latencies.append(response - prev_response)
-            prev_response = response
-    return first_token_latencies, token_to_token_latencies
+        total_time = request["response_timestamps"][-1] - request["timestamp"]
+        time_s = total_time / 1_000_000_000  # sec
+        time_ms = total_time / 1_000_000  # msec
+        end_to_end_latencies.append(time_ms)
+        throughputs.append(sequence_len / time_s)
+    return throughputs, end_to_end_latencies
 
 
-def calculate_avg_latencies(filename):
-    """Calculate avg first-token and avg total token-to-token latencies."""
+def calculate_offline_metrics(args, profile_result, filename):
+    """Calculate offline metrics that show end-to-end performance."""
     requests = load_json_data(filename)
-    first_token_latencies, token_to_token_latencies = collect_latencies(requests)
+    throughputs, end_to_end_latencies = collect_offline_metrics(
+        requests=requests, sequence_len=profile_result.prompt_size + args.max_tokens
+    )
 
-    # Compute mean and convert from nanosec to sec
-    avg_first_token_latency = np.mean(first_token_latencies) / 1_000_000_000
-    if token_to_token_latencies:
-        avg_token_to_token_latency = np.mean(token_to_token_latencies) / 1_000_000_000
-    else:
-        avg_token_to_token_latency = None
-    return avg_first_token_latency, avg_token_to_token_latency
+    profile_result.max_e2e_latency = max(end_to_end_latencies)
+    profile_result.min_e2e_latency = min(end_to_end_latencies)
+    profile_result.avg_e2e_latency = np.mean(end_to_end_latencies)
+    profile_result.max_token_throughput = max(throughputs)
+    profile_result.min_token_throughput = min(throughputs)
+    profile_result.avg_token_throughput = np.mean(throughputs)
+    profile_result.p50_token_throughput = np.percentile(throughputs, 50, method="lower")
+    profile_result.p90_token_throughput = np.percentile(throughputs, 90, method="lower")
+    profile_result.p95_token_throughput = np.percentile(throughputs, 95, method="lower")
+    profile_result.p99_token_throughput = np.percentile(throughputs, 99, method="lower")
 
 
 def summarize_profile_results(args, prompts):
     results = []
     for prompt in prompts:
         prompt_size = len(prompt.split())
         export_file = get_export_filename(args, prompt_size)
-        avg_first_token_latency, avg_total_t2t_latency = calculate_avg_latencies(
-            filename=export_file
-        )
 
-        profile_result = ProfileResults(
-            prompt_size=prompt_size,
-            avg_first_token_latency=avg_first_token_latency,
-            avg_total_t2t_latency=avg_total_t2t_latency,
-        )
+        profile_result = ProfileResults(prompt_size=prompt_size)
+        calculate_offline_metrics(args, profile_result, export_file)
```

Comment:
I think it might be a little more readable if this were changed to be like `calculate_metrics()`

then inside it would be like

```python
def calculate_metrics(...):
    calculate_always_needed_metric_1()
    calculate_always_needed_metric_2()
    if args.stream == False:
        calculate_sometimes_needed_metric_1()
```